yarn-project/foundation/src/abi/decoder.ts

```diff
@@ -57,14 +62,17 @@ class ReturnValuesDecoder {
 
   /**
    * Decodes all the return values for the given function ABI.
+   * Noir support only single return value (value can be struct).
    * @returns The decoded return values.
    */
-  public decode() {
-    const returnValues = [];
-    for (let i = 0; i < this.abi.returnTypes.length; i += 1) {
-      returnValues.push(this.decodeReturn(this.abi.returnTypes[i]));
+  public decode(): DecodedReturn {
+    if (this.abi.returnTypes.length > 1) {
+      throw new Error('Multiple return values not supported');
```

Comment:
shouldn't this be `different return types from one function is not supported` since you are checking `returnTypes.length`?

Reply:
Update - add a comment how an array is not `[Field, Field]` but `[array{size:2, type:field}]`